src/__tests__/App.test.tsx

```diff
@@ -0,0 +1,137 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { describe, beforeEach, it, expect, vi } from 'vitest';
+import * as api from '../api/api';
+import App from '../App';
+import ErrorBoundary from '../components/ErrorBoundary';
+
+const mockBooks = {
+  docs: [
+    { key: 'book1', title: 'Book One', author_name: ['Author One'] },
+    { key: 'book2', title: 'Book Two', author_name: ['Author Two'] },
+  ],
+  numFound: 2,
+};
+
+describe('App Component', () => {
+  beforeEach(() => {
+    vi.restoreAllMocks();
+    localStorage.clear();
+  });
+
+  it('makes initial API call on component mount', async () => {
```

Comment:
For consistency and readability, it’s better to use the same naming pattern _for all tests_ in the project.
Please choose your preferred style `should` or `action word`